backend/user-manager/src/index.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as admin from 'firebase-admin';
+import * as express from 'express';
+import * as morgan from 'morgan';
+import {check, validationResult} from 'express-validator';
+
+const app = express();
+app.use(morgan('common'));
+
+app.get(
+  '/api/users',
+  async (request: express.Request, response: express.Response) => {
+    try {
+      const result = await admin.auth().listUsers();
+      response.send(result.users);
+    } catch (e) {
+      response.status(500).send(e);
+    }
+  }
+);
+
+app.get(
+  '/api/users/:uid',
+  async (request: express.Request, response: express.Response) => {
+    const uid = request.params.uid;
+    try {
+      const user = await admin.auth().getUser(uid);
+      return response.send(user);
+    } catch (e) {
+      if (e.code === 'auth/user-not-found') {
+        return response.status(404).send(e);
+      }
+      return response.status(500).send(e);
+    }
+  }
+);
+
+app.put(
+  '/api/users/:uid',
+  [
+    check('uid').isLength({min: 20}),
+    check('role').isIn(['', 'worker', 'admin']),
+  ],
```

Comment:
If possible, this check should be done using an Istio AuthorizationPolicy. I'll let @josueetcom comment on how to implement that, however.

Reply:
this is validating the value to set as `role`. Data validation should not be in Istio, right?

Reply:
Ah I see, I misunderstood what this was doing. In that case, I believe there needs to be an AuthorizationPolicy on this service to prevent just anybody from being able to call this `PUT` method, right?

Reply:
yeah, i assume the whole thing will be behind istio auth, so I didn't add any token validation stuff here.

Reply:
Sorry, just to clarify, I am not suggesting to replace this validation here with Istio. They are separate concerns, and I was confused about what this code is doing in the first place.